### ISSUE TITLE
NPE when adding programmatically created subcommands to CommandLine

### ISSUE THREAD

**Issue by Mikusch** [remkop/picocli#381]
Hello,

I had a confusing NPE when trying to add a programmatically created subcommand using ``CommandLine#addSubCommand(String, Object)``.
```
picocli.CommandLine$InitializationException: Unable to initialize @ParentCommand field: java.lang.NullPointerException
	at picocli.CommandLine$Model$CommandReflection.initParentCommand(CommandLine.java:4186)
	at picocli.CommandLine.addSubcommand(CommandLine.java:234)
Caused by: java.lang.NullPointerException
	at picocli.CommandLine$Model$CommandReflection.initParentCommand(CommandLine.java:4175)
```
I took a look and it's because ``CommandLine#addSubCommand(String, Object)`` calls the ``userObject()`` method on the parent and subcommand, which returms null for both. Then the ``CommandReflection#initParentCommand(Object, Object)`` method tries calling ``getClass()`` on one of the null values, creating the NPE.

I can get around this issue using ``CommandLine.getSubcommands()`` and modifying the map it returns or simply adding the commands on the ``CommandSpec`` object instead.

Can you confirm this as a bug or is this intended behavior I just don't understand?

Kind regards

**Comment by remkop** [comment]
Hi, thanks for the bug report!
I have tried but I am unable to reproduce the problem. Can you share a program that demonstrates the issue?

**Comment by remkop** [comment]
Never mind, I got it now:
```
@Test
public void test381_NPE_whenAddingSubcommand() {
    CommandSpec toplevel = CommandSpec.create();
    toplevel.addOption(OptionSpec.builder("-o").description("o option").build());
    CommandLine commandLine = new CommandLine(toplevel);

    CommandSpec sub = CommandSpec.create();
    sub.addOption(OptionSpec.builder("-x").description("x option").build());
    commandLine.addSubcommand("sub", sub); // NPE here

    commandLine.usage(System.out);
}
```

**Comment by remkop** [comment]
I will try to do a bugfix release for this later today.
If you have additional tests please share.

**Comment by remkop** [comment]
I released [picocli v3.0.2](https://github.com/remkop/picocli/releases/tag/v3.0.2) with the fix. Should be available from Bintray and Maven soon.

Thanks again for the bug report!
